src/read.rs

```diff
@@ -223,26 +324,65 @@ impl<R: Read + io::Seek> ZipArchive<R> {
         self.names_map.keys().map(|s| s.as_str())
     }
 
+    /// Search for a file entry by name, decrypt with given password
+    pub fn by_name_decrypt<'a>(
+        &'a mut self,
+        name: &str,
+        password: &[u8],
+    ) -> ZipResult<ZipFile<'a>> {
+        self.by_name_internal(name, Some(password))
+    }
+
     /// Search for a file entry by name
     pub fn by_name<'a>(&'a mut self, name: &str) -> ZipResult<ZipFile<'a>> {
+        self.by_name_internal(name, None)
+    }
+
+    fn by_name_internal<'a>(
```

Comment:
I'd rather call this `by_name_with_optional_password`

Reply:
Okay. It's a bit long, but the name is clearer. Changed it to `by_name_with_optional_password` and `by_index_with_optional_password`.